pending/telegram_desktop.xml

```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    BleachBit
+    Copyright (C) 2008-2019 Andrew Ziem
+    https://www.bleachbit.org
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+    @app Telegram Desktop
+    @url https://desktop.telegram.org/
+    @os Windows, Linux, macOS
+    @cleanerversion v1.0.0
+    @cleanerdate 2019-11-14
+    @cleanerby Tobias B. Besemer
+    @tested ok v???, Windows ???
+    @testeddate ???
+    @testedby ???
+    @note Linux $$ProgramFiles$$ not yet figured out! I will do this later! Tobias.
+
+-->
+<cleaner id="telegram_desktop">
+  <label>Telegram Desktop</label>
+  <description>Messaging program</description>
+  <!-- Not yet figured out:
+  <running type="exe" os="linux">telegram</running>
+  -->
+  <running type="exe" os="windows">Telegram</running>
+  <option id="log">
+    <label>Log</label>
+    <description>Delete the log</description>
+    <!-- Linux: -->
+    <!-- Not yet figured out! -->
+    <!-- Windows: -->
+    <action command="delete" search="file" path="%AppData%\Telegram Desktop\log.txt"/>
+    <!-- macOS: -->
+    <!-- Not yet figured out! -->
+  </option>
+  <option id="memory_dumps">
+    <label>Memory Dumps</label>
+    <description>Delete the files</description>
+    <!-- Linux: -->
+    <!-- Not yet figured out! -->
+    <!-- Windows: -->
+    <action command="delete" search="walk.all" path="%AppData%\Telegram Desktop\tdata\dumps"/>
+    <!-- macOS: -->
+    <!-- Not yet figured out! -->
+  </option>
+  <option id="cache">
+    <label>Cache files</label>
+    <description>Delete the cache files</description>
+    <!-- Linux: -->
+    <!-- Not yet figured out! -->
+    <!-- Windows: -->
+    <action command="delete" search="walk.all" path="%AppData%\Telegram Desktop\tdata\user_data\cache"/>
+    <action command="delete" search="glob" path="%AppData%\Telegram Desktop\tdata\user_data\cache\*"/>
```

Comment:
This glob line looks redundant to the walk.all above it

Reply:
Hadn't we noted in the docs, that if I want to delete the content of a folder including all sub-folders, I have to use ATM "walk.all" followed by "glob"? Read that so in the doc, again...
This was AFAIK also the reason you introduced "walk.top" in BB3.0...
I noted in the doc to it:
> Starting with BleachBit v3.0 there is command="delete" search="walk.top" as a single command to do that!
> But as some users still stay on Windows XP with BleachBit v2.2 (last version for WinXP) but use new cleaners,
> it is highly recommended to use this new feature at the moment with care!

Reply:
To accomplish this goal, change `glob` to `file` and change `cache\*` to just `cache`. As it is, I think it won't delete the `cache` directory itself.

Reply:
AFAIK, my solution deletes all in the "cache" folder and your solution would delete the "cache" folder itself...

Reply:
I searched in "stable" for an example, but can't find one... The only example I found now (first hit) is this one: https://github.com/az0/cleanerml/blob/master/pending/windows_recycle_bins_more.xml

Reply:
The action there `search="glob" path="C:\$Recycle.Bin"` has no wildcard, so it is equivalent to `search="file"`.  

It might generate a lint warning because it has no wildcard.

Reply:
Here is the doc to it: https://github.com/LV-Crew/bleachbit-docus/blob/CleanerML-v2.0-Docu/_posts/cleanerml_v2.0.md#407-delete-the-content-of-a-folder-but-not-the-folder-itself

Reply:
If you want to delete the folders in the folder but not the root folder, then I think you need just one `walk.all` action. Documentaton:
https://docs.bleachbit.org/cml/cleanerml.html

>walk.all: matches all and directories files under a directory.

Reply:
And I fear I used a bad example with the bin cleaner because I forgot there a "\*" at the end...

Reply:
AFAIK was the prob, that "walk.all" left the sub-folder structure back...

Reply:
Could be a bug?

Reply:
So what's about search="walk.top" ??? I just still avoid to use it...

Reply:
...there was a reason why you introduced it in BB3.0...

Reply:
Windows XP support ended April 2014, five and a half years ago, and there are not many people still using it. Considering finite resources, it's not a priority

Reply:
...but means, if I change it, I should use "walk.top", right?

What ever... I can also make a test run of the cleaner and have a look if he deletes as I planed it... but I need a break! Was working the whole night on PRes... I check it later!

Reply:
If you want to delete the top directory, I would use `walk.top`.  

Yes, feel free to take a break! You work faster than I do :)

Reply:
Thx for the compliment! :-) But the harder Python part still remains to you... ;-)

Yeah, seems that's the prob! I just want to delete the complete content incl. the sub-folders of the folder!
And I think I can remember that we had a other solution to accomplish this goal, too, but with the solution I documented and used now, the cleaning of the files is better displayed and/or calculated in BB because BB deletes them all step by step and displays and calculates it! ;-)

I make now a test run with the cleaner and makes some screenshot of the folder & content & BB output during this for you. :-)

Reply:
Here some SS...

The folder "cache" have just one sub-folder in my case...

![2019-11-14 07_24_49-C__Users_Administrator_AppData_Roaming_Telegram Desktop_tdata_user_data_cache](https://user-images.githubusercontent.com/11881073/68835768-15a3e380-06b9-11ea-9253-afd44f2b348c.jpg)

...but in this sub-folder there are 257 (!!!) more sub-folders that include the >15.500 files !!!

![2019-11-14 07_25_08-C__Users_Administrator_AppData_Roaming_Telegram Desktop_tdata_user_data_cache_0](https://user-images.githubusercontent.com/11881073/68835838-42f09180-06b9-11ea-950e-3b74a9082d29.jpg)

First a test of deleting the log file...

![2019-11-14 08_18_09-BleachBit](https://user-images.githubusercontent.com/11881073/68835887-5f8cc980-06b9-11ea-9dc3-eef3a63c4f51.jpg)

...then a test of deleting the "Memory Dumps"...

![2019-11-14 08_21_11-BleachBit](https://user-images.githubusercontent.com/11881073/68835972-8fd46800-06b9-11ea-8cbc-1e1c325ff08f.jpg)

...and then the cleaning of the cache folders...
...but as you can see: It run into an Error 1 !!!

![2019-11-14 08_35_01-BleachBit](https://user-images.githubusercontent.com/11881073/68836097-d75af400-06b9-11ea-89bf-b734e9e8af69.jpg)

Here is a SS from the debugging window:

![2019-11-14 08_36_34-Administrator_ C__Windows_system32_cmd exe - bleachbit_console exe  --gui --debu](https://user-images.githubusercontent.com/11881073/68836176-096c5600-06ba-11ea-96e4-489e77081457.jpg)

But the cache folder is empty as intended !!!

![2019-11-14 08_38_25-C__Users_Administrator_AppData_Roaming_Telegram Desktop_tdata_user_data_cache](https://user-images.githubusercontent.com/11881073/68836261-3a4c8b00-06ba-11ea-97e1-4bb5ac3a40dc.jpg)

...so I have no clue what the "Error 1" should mean!

Note: There are maybe some space for improvements of the cleaner regarding the strings used...

Reply:
Sorry, I don't understand

Reply:
OK, which part?

The cleaner works like intended, but I get a error 1.